scripts/pylib/twister/twisterlib.py

```diff
@@ -3366,39 +3370,38 @@ def load_quarantine(self, file):
             self.quarantine.update(d)
 
     def load_from_file(self, file, filter_status=[], filter_platform=[]):
-        try:
-            with open(file, "r") as fp:
-                cr = csv.DictReader(fp)
-                instance_list = []
-                for row in cr:
-                    if row["status"] in filter_status:
-                        continue
-                    test = row["test"]
-
-                    platform = self.get_platform(row["platform"])
-                    if filter_platform and platform.name not in filter_platform:
-                        continue
-                    instance = TestInstance(self.testcases[test], platform, self.outdir)
-                    if self.device_testing:
-                        tfilter = 'runnable'
-                    else:
-                        tfilter = 'buildable'
-                    instance.run = instance.check_runnable(
-                        self.enable_slow,
-                        tfilter,
-                        self.fixtures
-                    )
-                    instance.create_overlay(platform, self.enable_asan, self.enable_ubsan, self.enable_coverage, self.coverage_platform)
-                    instance_list.append(instance)
-                self.add_instances(instance_list)
-
-        except KeyError as e:
-            logger.error("Key error while parsing tests file.({})".format(str(e)))
-            sys.exit(2)
-
-        except FileNotFoundError as e:
-            logger.error("Couldn't find input file with list of tests. ({})".format(e))
-            sys.exit(2)
+        with open(file, "r") as json_test_plan:
+            jtp = json.load(json_test_plan)
+            instance_list = []
+            for ts in jtp.get("testsuites", []):
+                logger.debug(f"loading {ts['name']}...")
+                #if ts["status"] in filter_status:
```

Comment:
seems that this is a typo, I think we should enable filter by test instance status.

Reply:
it is being done differently now, we no longer modify existing files. But there were issues that have been addressed.